system/utils.py

```diff
@@ -205,28 +262,99 @@ def wrapped(operations, queue_size, add_size, get_size, event_loop):
     return wrapped
 
 
-async def send_and_get_nym(pool_handle, wallet_handle, trustee_did, some_did):
+async def send_and_get_nym(pool_handle, wallet_handle, trustee_did, some_did=None):
+    if some_did is None:
+        some_did, _ = await did.create_and_store_my_did(wallet_handle, '{}')
     add = await write_eventually_positive(send_nym, pool_handle, wallet_handle, trustee_did, some_did)
     assert add['op'] == 'REPLY'
     get = await read_eventually_positive(get_nym, pool_handle, wallet_handle, trustee_did, some_did)
     assert get['result']['seqNo'] is not None
 
 
-async def check_ledger_sync():
-    hosts = [testinfra.get_host('ssh://node{}'.format(i)) for i in range(1, 8)]
+# TODO make that async
+def check_no_failures(hosts):
+    for host in hosts:
+        result = host.run('journalctl -u indy-node.service -b -p info')
+        assert result.find("indy-node.service: Failed") == -1, (
+            "Node service on host{} failed:\n{}".format(host.id, result)
+        )
+
+# TODO make that async
+async def check_ledger_sync(node_ids=None, nodes_num=7):
+    if node_ids is None:
+        node_ids = [(i + 1) for i in range(nodes_num)]
+    hosts = [NodeHost(i) for i in node_ids]
+
     pool_results = [host.run('read_ledger --type=pool --count') for host in hosts]
-    print('\nPOOL LEDGER SYNC: {}'.format([result.stdout for result in pool_results]))
+    print('\nPOOL LEDGER SYNC: {}'.format([result for result in pool_results]))
```

Comment:
Original output was more readable, however I do understand your concerns that stderr and rc were missing. What about creating some helper function that always outputs stdout, outputs stderr if it is not empty and return code if it is non zero?